### PR TITLE
Platform - UI: Login page

### PR DESCRIPTION
**Pull request by deetz99**
*Issue:*

- bcgov/entity#24116

*Description of changes:*

generic login page and auth middleware

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the strr license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-platform-dev--pr-248-nqww36q0.web.app

**Review by thorwolpert — APPROVED** [review]

File: strr-platform-web/app/pages/auth/login.vue [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<script setup lang="ts">
+const { t, locale } = useI18n()
+const keycloak = useKeycloak()
+const redirectUrl = useRuntimeConfig().public.baseUrl + locale.value + '/platform/dashboard'
```
Comment (kialj876) [inline strr-platform-web/app/pages/auth/login.vue:4]: I updated 1password - can you please update the NUXT_BASE_URL inside `vaults.env` to grab from ...`strr/STRR_PLATFORM_URL`

File: strr-platform-web/app/pages/auth/login.vue
Comment (kialj876) [inline strr-platform-web/app/pages/auth/login.vue:1]: should this page check to see if they are already logged in, and if so then redirect them to 'redirectUrl'? Probably a specific middleware thing for this page

**Review by kialj876 — COMMENTED** [review]
Looks good to me, just a couple questions

File: strr-platform-web/app/pages/auth/login.vue [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+keycloak.setLogoutRedirectUrl(logoutUrl)
+
+// page stuff
+useHead({
+  title: t('platform.title.login')
+})
+
+setBreadcrumbs([
+  { label: t('label.bcregDash'), to: useRuntimeConfig().public.registryHomeURL + 'dashboard' }
+])
```
Comment (kialj876) [inline strr-platform-web/app/pages/auth/login.vue:16]: do you think this should have a 'login' crumb too?

File: strr-platform-web/app/pages/auth/login.vue [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          block
+          :ui="{
+            gap: { sm: 'gap-x-2.5' }
+          }"
+          @click="keycloak.login(IdpHint.BCEID, redirectUrl)"
+        />
+        <UButton
+          :label="$t('label.loginBcsc')"
+          icon="i-mdi-account-card-details-outline"
+          color="gray"
```
[the remark this answers is not in the document]
Comment (deetz99) [reply strr-platform-web/app/pages/auth/login.vue:38]: Yes that is the idea, try to push people to login with BCeID

File: strr-platform-web/app/pages/auth/login.vue
Comment (deetz99) [reply strr-platform-web/app/pages/auth/login.vue:1]: Yes I can add this in.

File: strr-platform-web/app/pages/auth/login.vue (hunk 2 above)
Comment (deetz99) [reply strr-platform-web/app/pages/auth/login.vue:16]: Probably a good idea, should the first breadcrumb link go to the registry dashboard still? or housing? I could add a todo comment instead for now

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-platform-dev--pr-248-nqww36q0.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-platform-dev--pr-248-nqww36q0.web.app

File: strr-platform-web/app/assets/styles/layout.css
Comment (deetz99) [inline strr-platform-web/app/assets/styles/layout.css:1]: had to add this to allow grow in the pages in order to centre the login card

**Review by kialj876 — APPROVED** [review]
Nice work!

**Pull request merged**